### PR TITLE
Show no-session alert

### PR DESCRIPTION
**Pull request by leftmostcat**
### Problem

There's no indication on the site when redis isn't up.

### Solution

Add scaffolding to show alerts on the site, use this to display a message when we can't find a session.

### Areas of Impact

Page display.

### CODE REVIEW

File: src/client/components/pages/contribute.js [hunk 1]
```diff
@@ -58,7 +58,7 @@ function ContributePage() {
 					BookBrainz style guidelines
 				</a> when adding data.
 				Unfortunately the aforementioned are still work in progress
-				and should they not cover some cases, please refer to the
+				and should they not cover some cases, please refer to the&nbsp;
```
Comment (leftmostcat) [inline src/client/components/pages/contribute.js]: Opportunistic fix to the Contributing page I found while testing.

File: src/server/helpers/props.js [hunk 2]
```diff
@@ -27,5 +27,5 @@ export function escapeProps(props) {
 }
 
 export function generateProps(req, res, props) {
-	return Object.assign({}, req.app.locals, res.locals, props);
+	return Object.assign({}, res.locals, props);
```
Comment (leftmostcat) [inline src/server/helpers/props.js]: I took this opportunity to pull out `req.app.locals`. There's a ton of information in there that's not relevant to the user and I don't want to risk anything sensitive being added.

Comment (Arthelon) [reply src/server/helpers/props.js]: have you tested this out on all routes? I believe there were some request variables that were being used.

Comment (leftmostcat) [reply src/server/helpers/props.js]: I haven't tested it on all routes, but as far as I can tell, the only things which were being pulled from there were site revision and git repo URL. I've moved both of these into `res.locals`.

File: src/client/containers/layout.js [hunk 3]
```diff
@@ -191,6 +191,15 @@ class Layout extends React.Component {
 			</div>
 		);
 
+		const alerts = [];
```
Comment (LordSputnik) [inline src/client/containers/layout.js]: Can we use:

    const alerts = this.props.alerts.map(...)

I think that's probably a bit cleaner here.

File: src/server/app.js [hunk 4]
```diff
@@ -112,6 +112,14 @@ app.use((req, res, next) => {
 	res.locals.repositoryUrl = repositoryUrl;
 	res.locals.alerts = [];
 
+	if (!req.session) {
+		res.locals.alerts.push({
+			level: 'danger',
+			message: 'We are currently experiencing technical difficulties; ' +
+				'logins will not work until this is resolved.'
```
Comment (LordSputnik) [inline src/server/app.js]: Maybe "signing in will not work until this is resolved." makes more sense?

File: src/client/containers/layout.js (hunk 3 above)
Comment (leftmostcat) [reply src/client/containers/layout.js]: Good catch.

File: src/server/app.js (hunk 4 above)
Comment (leftmostcat) [reply src/server/app.js]: Will address.

**Pull request merged**